File: medilink_client/lib/models/feedback_response.dart


class FeedbackResponse {
  String? id;
  String? feedback;
  String? response;

  FeedbackResponse({
    this.id,
    this.feedback,
    this.response,
  });

  factory FeedbackResponse.fromJson(Map<String, dynamic> json) {
    return FeedbackResponse(
      id: json['_id'] as String?,
      feedback: json['feedback'] as String?,
      response: json['response'] as String?,
    );
  }
}

File: back_end/models/feedback_response.js
const mongoose = require('mongoose');


const feedbackResponseSchema = new mongoose.Schema({
  feedback: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Feedback',
    required: true,
  },
  response:{
    type:String,
    required:true
  }
});

const FeedbackResponse = mongoose.model('FeedbackResponse', feedbackResponseSchema);

module.exports = FeedbackResponse;
